yarn-project/acir-simulator/src/client/simulator.ts

```diff
@@ -144,37 +144,38 @@ export class AcirSimulator {
     storageSlot: Fr,
     notePreimage: Fr[],
   ) {
-    const abi = await this.db.getFunctionABI(contractAddress, computeNoteHashAndNullifierSelector);
-    if (!abi) {
+    try {
+      const abi = await this.db.getFunctionABI(contractAddress, computeNoteHashAndNullifierSelector);
+
+      const preimageLen = (abi.parameters[3].type as ArrayType).length;
```

Comment:
Is there no cleaner cast we can do for this parameter type?

Reply:
I have just fixed the issue with error here so I have not written the code. @LeilaWang might know better.